### PR TITLE
recipe(mms-1b-all): add verified CPU fp32/fp16 configs

### PR DESCRIPTION
**Pull request by fuliucansheng**
## Summary

Adds verified CPU FP32 and FP16 recipes for `facebook/mms-1b-all` automatic speech recognition.

This PR was fully rerun after generalized Wav2Vec2 support landed on `main`. Current `main` already resolves `Wav2Vec2ForCTC` to `AutoModelForCTC`, so the old model-specific mapping and its tests were removed. The final diff is recipe-only.

## Model

- Checkpoint: `facebook/mms-1b-all`
- Architecture: `Wav2Vec2ForCTC`, 964.8M parameters, 48 encoder layers
- Task: `automatic-speech-recognition`
- Contract: `input_values float32 [1,16000] -> logits float32 [1,49,154]`
- Tested base: `8079522ee692645569508d11abd4a0ea45a40987`
- Tested head: `b91152fb07a8a769a4756e94006bf83ede7073c9`
- Environment: winml `0.2.0`, Transformers `5.14.1`, PyTorch `2.13.0`, ORT `1.27.0`

## Why recipe-only

Clean-main `winml inspect` selected:

```text
task: automatic-speech-recognition
model_type: wav2vec2
hf_model_class: AutoModelForCTC
source: TasksManager
```

A clean-main recipe-free FP32 config/build also loaded `Wav2Vec2ForCTC` and completed successfully. The shared resolver now passes `model_type=wav2vec2` to Optimum and already has a Wav2Vec2-versus-Whisper regression. No custom mapping is required.

The recipes were regenerated under Transformers 5 and retain:

```json
"compatibility": {
  "transformers_attention": "eager"
}
```

## Validation

| Tier | Tuple | Result |
|---|---|---|
| L0 | CPU / FP32 | PASS, build 546.1s |
| L0 | CPU / FP16 | PASS, build 596.5s |
| L1 | CPU / FP32 | 301.452 ms mean, 3.32 samples/s |
| L1 | CPU / FP16 | 335.36 ms mean, 2.98 samples/s |
| L2 | CPU / FP32 | cosine 0.999999999999424, max abs 0.00010419, frame argmax 49/49 |
| L2 | CPU / FP16 | cosine 0.9999999554162311, max abs 0.02493513, frame argmax 49/49 |
| L3 | CPU / FP32 | one pinned real-audio functional smoke decoded non-empty text |

L3 is intentionally a functional smoke only. It does not claim WER or representative ASR accuracy.

### Precision and size

| Artifact | Weight dtype | External weight bytes | Nodes |
|---|---|---:|---:|
| FP32 | 1,096 FLOAT | 3,859,383,684 | 1,921 |
| FP16 | 1,096 FLOAT16 | 1,929,691,842 | 1,923 |

FP16 preserves FP32 I/O and adds only two boundary `Cast` nodes. It halves weight storage but was 11.2% slower than FP32 on this CPU; no CPU speedup is claimed.

### Functional smoke

- Source: one public `Narsil/asr_dummy` real-speech fixture
- Source SHA-256: `30885601173f96b0d8ddd020dc959b055c6c1582b85a33e3fcab8c4b08ed94c2`
- Deterministic window: highest-RMS 16,000-sample segment at 16 kHz
- Shape: `[1,16000] -> [1,49,154]`
- Decoded text: `he hoped there would be`

`winml eval --mode compare` currently rejects `automatic-speech-recognition`, and `winml run` is disabled. The smoke therefore used the checkpoint's `AutoProcessor`, the final ORT artifact, argmax CTC tokens, and the checkpoint decoder. This preserves the blocker honestly and does not invent an accuracy metric. Eval support remains tracked by #1229.

## Component and operator analysis

The hierarchy-preserving graph accounts exactly for:

- 48 encoder layers
- 386 `Gemm`: feature projection 1 + encoder 48 x 8 + LM head 1
- 153 `LayerNormalization`: feature extractor 7 + projection 1 + encoder 48 x 3 + final 1
- 96 `MatMul`, 48 `Softmax`, 48 `Mul`, and 48 `Relu`

Static rules were run for all populated targets in the current ModelKitArtifacts rules tree: OpenVINO CPU/GPU/NPU and QNN GPU/NPU. No operator type was explicitly partial or unsupported, but all 13 FP32 / 14 FP16 operator types were classified `unknown`; this is rule-data evidence, not accelerator runtime support. QNN reported missing rules for `Gelu` and `Gemm`, plus FP16 NPU `LayerNormalization` shape inference.

## Quality gates

```text
48 passed
  tests/unit/loader/test_config_resolution.py
  tests/unit/loader/test_resolve_task_and_model_class.py
  tests/unit/eval/test_recipes.py

recipe JSON/loader/precision/Transformers-compatibility contract: PASS
git diff --check: PASS
```

## Reproduce

```powershell
winml inspect -m facebook/mms-1b-all --task automatic-speech-recognition --format json

winml build `
  -m facebook/mms-1b-all `
  -c examples/recipes/facebook_mms-1b-all/cpu/cpu/automatic-speech-recognition_fp32_config.json `
  -o out/fp32 --device cpu --precision fp32

winml build `
  -m facebook/mms-1b-all `
  -c examples/recipes/facebook_mms-1b-all/cpu/cpu/automatic-speech-recognition_fp16_config.json `
  -o out/fp16 --device cpu --precision fp16

winml perf -m out/fp32/model.onnx --device cpu --ep cpu --warmup 1 --iterations 3
winml perf -m out/fp16/model.onnx --device cpu --ep cpu --warmup 1 --iterations 3
```

### CODE REVIEW

**Comment by ssss141414** [comment]
https://github.com/microsoft/winml-cli/pull/1095
or
https://github.com/microsoft/winml-cli/pull/1154 
may resolve this question. Please wait for merge and see if we only need recipes...

**Review by ssss141414 — APPROVED** [review]
Approved at exact tested head b91152fb07a8a769a4756e94006bf83ede7073c9. Full rerun on current main confirmed generalized AutoModelForCTC resolution, fresh FP32/FP16 builds, CPU perf, deterministic PyTorch parity, one bounded real-audio functional smoke, component/operator analysis, 48 focused tests, and 9/9 GitHub checks. Final diff is recipe-only.

**Review by ssss141414 — APPROVED** [review]

**Review by DingmaomaoBJTU — APPROVED** [review]
